filebeat/input/filestream/input_integration_test.go

```diff
@@ -0,0 +1,144 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build integration
+
+package filestream
+
+import (
+	"context"
+	"runtime"
+	"sync"
+	"testing"
+
+	loginp "github.com/elastic/beats/v7/filebeat/input/filestream/internal/input-logfile"
+	input "github.com/elastic/beats/v7/filebeat/input/v2"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/logp"
+)
+
+// test_close_renamed from test_harvester.py
+func TestFilestreamCloseRenamed(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("renaming files while Filebeat is running is not supported on Windows")
+	}
```

Comment:
Really? I wonder if the issue is due to use trying to write to the "original" path after the rotation.

Reply:
I got this error when I ran `os.Rename`: `The process cannot access the file because it is being used by another process.`

I think no because the test tried to rename the file after it had written lines to the file. It rather fails because the `filestream` input keeps the file open as it is waiting for new events to show up. To test my theory I closed the file on EOF, afterwards, I was able to rename the file on Windows as well.

Reply:
When reopening restarting the input, is that detected as a rename? Can we create a test to validate the expected behavior?

Reply:
If someone is able to rename a file on Windows while Filebeat is keeping it open, it is detected by the checker goroutine of `logReader`. So the reader is going to behave correctly.

> When reopening restarting the input, is that detected as a rename? Can we create a test to validate the expected behavior?

What is the expected behaviour? If the input is restarted, it does not make sense to check if a file was renamed because `close_renamed` applies to opened files that are renamed during Filebeat running a `Harvester` for.

When the input is restarted, it finds the state in the registry for the renamed file, unless `path`  `file_identity` strategy is selected, it will not start a new `Harvester` for the file because it is still the same file it encountered during the previous run.

These test cases are not special, we either already have tests for it or going to have them when all tests of `log` input is ported to use `filestream`.

Reply:
I still would expect that the meta-data in the registry are updated... meaning that the prospector detects the file was renamed. Maybe it is rather worth an unit test for the prospector

Reply:
OK, I will address this in a separate PR.